### PR TITLE
LPS-185533 Convert the Optional Dynamic Greedy references to static references using Snapshot in module portal-search-solr8-impl

### PR DESCRIPTION
**Pull request by haydenzheng314**
Ticket: https://issues.liferay.com/browse/LPS-185533
Thanks

### CODE REVIEW

**Comment by LisztLiszt** [comment]
ci:help

**Comment by LisztLiszt** [comment]
ci:test:search-solr

**Comment by LisztLiszt** [comment]
ci:test:startup

**Comment by zhangyan0701** [comment]
ci:test:search-solr

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/test/java/com/liferay/portal/search/solr8/internal/SolrIndexingFixture.java [hunk 1]
```diff
@@ -257,7 +257,7 @@ protected NGramQueryBuilderImpl createNGramQueryBuilder() {
 			new NGramQueryBuilderImpl();
 
 		ReflectionTestUtil.setFieldValue(
-			nGramQueryBuilderImpl, "_nGramHolderBuilder",
+			nGramQueryBuilderImpl, "_nGramHolderBuilderSnapshot",
```
Comment (dantewang) [inline modules/apps/portal-search-solr8/portal-search-solr8-impl/src/test/java/com/liferay/portal/search/solr8/internal/SolrIndexingFixture.java:260]: @LisztLiszt This part is wrong. The field is now of type `Snapshot`, so you will fail to set a `NGramQueryBuilder` to it.

The correct way is to register the `nGramQueryBuilderImpl` using BundleContext. Please see Jiaxu's PR modifying ElasticSearch modules for example.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**